mmda/predictors/hf_predictors/span_group_classification_predictor.py

```diff
@@ -0,0 +1,358 @@
+        )
+        # since input data is automatically chunked into segments (e.g. 512 length),
+        # each example <dict> actually becomes many input sequences.
+        # this mapper unpacks all of this into one <dict> per input sequence.
+        # we set `repeat` because we want other fields (`context_id`) to repeat across sequnces
+        self.unpacking_mapper = UnpackingMapper(
+            fields_to_unpack=[
+                self._HF_RESERVED_INPUT_IDS,
+                self._HF_RESERVED_ATTN_MASK,
+                self._HF_RESERVED_WORD_IDS
+            ],
+            ignored_behavior='repeat'
+        )
+        # at the end of this, each <dict> contains <lists> of length `batch_size`
+        # where each element is variable length within the `max_length` limit.
+        # `keep_last` controls whether we want partial batches, which we always do
+        # for token classification (i.e. we dont want to miss anything!)
+        self.batch_size_mapper = FixedBatchSizeMapper(
+            batch_size=batch_size,
+            keep_last=True
+        )
+        # this performs padding so all sequences in a batch are of same length
+        self.list_collator_mapper = FromTokenizerListCollatorMapper(
+            tokenizer=tokenizer,
+            pad_to_length=None,  # keeping this `None` is best because dynamic padding
+            fields_pad_ids={
+                self._HF_RESERVED_WORD_IDS: self._HF_RESERVED_WORD_PAD_VALUE
+            }
+        )
+        # this casts python Dict[List] into tensors.  if using GPU, would do `device='gpu'`
+        self.python_to_torch_mapper = Python2TorchMapper(
+            device=device
+        )
+        # combining everything
+        self.preprocess_mapper = self.tokenizer_mapper >> \
+                                 self.unpacking_mapper >> \
+                                 self.batch_size_mapper
+
+    @classmethod
+    def from_pretrained(
+            cls,
+            model_name_or_path: str,
+            span_group_name: str,
+            context_name: str,
+            batch_size: Optional[int] = 2,
+            device: Optional[str] = 'cpu',
+            *args,
+            **kwargs
+    ):
+        """If `model_name_or_path` is a path, should be a directory
+        containing `vocab.txt`, `config.json`, and `pytorch_model.bin`
+
+        NOTE: slightly annoying, but if loading in this way, the `_name_or_path`
+        in `model.config` != `config`.
+        """
+        tokenizer = transformers.AutoTokenizer.from_pretrained(
+            pretrained_model_name_or_path=model_name_or_path, *args, **kwargs
+        )
+        config = transformers.AutoConfig.from_pretrained(
+            pretrained_model_name_or_path=model_name_or_path, *args, **kwargs
+        )
+        model = transformers.AutoModelForTokenClassification.from_pretrained(
+            pretrained_model_name_or_path=model_name_or_path, *args, **kwargs
+        )
+        predictor = cls(model, config, tokenizer,
```

Comment:
I would use kwargs for everything, i.e., 

```suggestion
        predictor = cls(model=model, config=config, tokenizer=tokenizer,
```

Reply:
done